src/core/document.js

```diff
@@ -401,15 +422,21 @@ class Page {
     const newAnnotationsByPage = !this.xfaFactory
       ? getNewAnnotationsMap(annotationStorage)
       : null;
+    const deletedAnnotations = new RefSet();
 
     let newAnnotationsPromise = Promise.resolve(null);
     if (newAnnotationsByPage) {
       const newAnnotations = newAnnotationsByPage.get(this.pageIndex);
       if (newAnnotations) {
```

Comment:
Nit: Maybe avoid creating an unnecessary `RefSet`-instance by default?
```suggestion
    let deletedAnnotations = null;

    let newAnnotationsPromise = Promise.resolve(null);
    if (newAnnotationsByPage) {
      const newAnnotations = newAnnotationsByPage.get(this.pageIndex);
      if (newAnnotations) {
        deletedAnnotations = new RefSet();
```